### PR TITLE
Add function to add http.Handler as Middleware

### LINKED ISSUE #52: Middleware adapter for http.Handler middleware
Lots of other projects use an http.Handler model for middleware, which we don't; I'd like to have an adapter to convert `http.Handler` to `middleware.Middleware`, probably by just running the provided handler inside our middleware signature. An option to reject on some condition after it is run would also be very helpful.

Update May 8 2023: There's a framework in 1.2.0 for this now, but we need to address the way that traditional middleware is constructed. Typically you get a `func (http.Handler) http.Handler`, where the resulting handler passes control to the argument handler on "success". We need to emulate this in our middleware model to ensure that things work as expected.

### PR DESCRIPTION
**Pull request by SuneelFreimuth**
Closes #52

### CODE REVIEW

**Comment by jakenichols2719** [comment]
Something went wrong with the versioning here. `middleware.ExecuteMiddleware` is getting removed, which is currently used to execute any of the native middleware that's attached. It looks like the merge of `v1.2.0` broke something, the upstream branch for that version has `middleware.Exec...` still.

**Comment by SuneelFreimuth** [comment]
Fixed.

File: pkg/middleware/middleware.go
Comment (jakenichols2719) [inline pkg/middleware/middleware.go:1]: Go convention does actually dictate that comments start with the function/struct name; I haven't done a great job of this in the past, but I'm working on moving comments over to that now. Not a requirement, and we need a doc pass anyways, so not concerned about it right now.

**Comment by jakenichols2719** [comment]
Everything looks good, thank you for the fix. There are additional things to do as noted in the issue, but having this laid out in 1.2.0 is a great start.

**Pull request merged**